BatchProcessing/BatchStructureSetConversion.py

```diff
@@ -321,16 +321,20 @@ def save_rtslices(output_dir, use_ref_image, ref_image_node_id=None):
     if exist_db:
       logging.info('BatchStructureSet running in existing database mode')
       DICOMUtils.openDatabase(input_folder)
-      all_patients = slicer.dicomDatabase.patients()
+      db = slicer.dicomDatabase
+      all_patients = db.patients()
       logging.info('Must Process Patients %s' % len(all_patients))
-
       for patient in all_patients:
-        slicer.mrmlScene.Clear(0) # clear the scene
-        DICOMUtils.loadPatientByUID(patient)
-        output_dir = os.path.join(output_folder,patient)
+        slicer.mrmlScene.Clear(0) # clear the scen
+        studyList=db.studiesForPatient(patient) # code modified from https://www.slicer.org/wiki/Documentation/Nightly/ScriptRepository 
+        seriesList=db.seriesForStudy(studyList[0])
+        fileList=db.filesForSeries(seriesList[0])
+        print(db.fileValue(fileList[0],'0010,0020'))
+        DICOMUtils.loadPatientByUID(patient)       
+        output_dir = os.path.join(output_folder,db.fileValue(fileList[0],'0010,0020'))
         if not os.access(output_dir, os.F_OK):
           os.mkdir(output_dir)
-        save_rtslices(output_dir, use_ref_image)

```

Comment:
Please add the argument back, this would break the reference image feature.

Reply:
What's the difference between "save_rtslices(output_dir, use_ref_image)" and "save_rtslices(output_dir)". I know this is a previous issue, but after reading the post, I can't realize what is reference image.

Reply:
If you specify a reference image, then the saved NRRDs will have that dimension instead of the segments being cropped to their "effective extent". But even if you don't understand a feature, removing it is not a good idea :)